astronomy-pipeline/extract_functions.py

```diff
@@ -0,0 +1,185 @@
+    """Reusable function for getting a database connection."""
+
+    return connect(cursor_factory=extras.RealDictCursor,
+                   dbname=ENV["DB_NAME"],
+                   host=ENV["DB_HOST"],
+                   password=ENV["DB_PASSWORD"],
+                   user=ENV["DB_USER"],
+                   port=ENV["DB_PORT"])
+
+
+def get_auth_string() -> str:
+    """Generates AstronomyAPI authorization key from env values."""
+
+    user_pass = f"{ENV["ASTRONOMY_ID"]}:{ENV["ASTRONOMY_SECRET"]}"
+
+    auth_string = base64.b64encode(user_pass.encode()).decode()
+
+    return auth_string
+
+
+def get_db_regions() -> list:
+    """Returns the regions data from an RDS instance"""
+
+    with get_db_connection() as conn:
+        cur = conn.cursor()
+
+        cur.execute("SELECT * FROM region;")
+
+        regions = cur.fetchall()
+
+    return regions
+
+
+def get_all_body_positions(start_date: date, end_date: date, lat: float,
+                           long: float, time: str, elev: int = 50) -> dict:
+    """Returns the inclusive body positional information from
+    the Astronomy API for a given date range."""
+
+    url = f"{ASTRO_URL}/bodies/positions?latitude={lat}&longitude={
+        long}&elevation={elev}&from_date={start_date}&to_date={end_date}&time={time}"
+
+    auth_string = get_auth_string()
+
+    headers = {"Authorization": f"Basic {auth_string}"}
+
+    return (requests.get(url, headers=headers, timeout=10)).json()
+
+
+def make_clean_body_dict(entry: dict) -> dict:
+    """Creates refined body data from astronomy API body data."""
+
+    clean_dict = {}
+    clean_dict["datetime"] = entry["date"]
+    clean_dict["body_name"] = entry["id"]
+    clean_dict["distance_km"] = entry["distance"]["fromEarth"]["km"]
+    clean_dict["azimuth"] = entry["position"]["horizontal"]["azimuth"]["degrees"]
+    clean_dict["altitude"] = entry["position"]["horizontal"]["altitude"]["degrees"]
+    clean_dict["constellation"] = entry["position"]["constellation"]["id"]
+
+    return clean_dict
+
+
+def refine_bodies_data(bodies: dict) -> list:
+    """Removes unnecessary data from the astronomy positions dict."""
+
+    rows = bodies["data"]["table"]["rows"]
+
+   # Each cell contains a list of dictionaries of an entry
+   # key and a single body over a set number of days (7)
+    output_list = []
+    for obj in rows:
+        for entry in obj["cells"]:
+            clean_dict = make_clean_body_dict(entry)
+            output_list.append(clean_dict)
+
+    return output_list
+
+
+def get_moon_phase(input_date: str) -> str:
+    """Returns a url for an image of the moon phase for a given
+    location and date."""
+
+    example_body = {
+        "format": "png",
+        "style": {
+            "moonStyle": "default",
+            "backgroundStyle": "stars",
+            "headingColor": "white",
+            "textColor": "white"
+        },
+        "observer": {
+            "latitude": 53.812207,
```

Comment:
The function's doc string says it takes in both location and date, but location is predefined? Is this intended, and to be added later or?

Reply:
Good spot, the function should only take date I just forgot to amend the Docstring. For location, as we are generalising moon phase I just picked a central location in the UK.

Reply:
Nice one, makes sense.